app/components/wallet/trezorConnect/SaveDialog.js

```diff
@@ -0,0 +1,189 @@
+            ]
+          )],
+        },
+      },
+    }, {
+      options: {
+        validateOnChange: true,
+        validationDebounceWait: 250,
+      },
+    });
+  }
+
+  render() {
+    const { intl } = this.context;
+
+    // walletNameBlock
+    const walletNameFieldClasses = classnames([
+      'walletName',
+      styles.walletName,
+    ]);
+    const walletNameField = this.form.$('walletName');
+
+    const walletNameBlock = (
+      <div className={classnames([styles.headerBlock, styles.headerSaveBlock])}>
+        <Input
+          className={walletNameFieldClasses}
+          {...walletNameField.bind()}
+          error={walletNameField.error}
+          skin={<SimpleInputSkin />}
+        />
+        <span>{intl.formatMessage(messages.saveWalletNameInputBottomInfo)}</span>
+      </div>);
+
+    // middleBlock selection depending upon state
+    let middleBlock = null;
+
+    switch (this.props.progressInfo.stepState) {
+      case StepStateOption.LOAD:
+        middleBlock = (
+          <div className={classnames([styles.middleBlock, styles.middleSaveLoadBlock])}>
+            <SvgInline svg={saveLoadGIF} cleanup={['title']} />
+          </div>);
+        break;
+      case StepStateOption.PROCESS:
+        // START
+        middleBlock = (
+          <div className={classnames([styles.middleBlock, styles.middleSaveStartProcessBlock])}>
+            <SvgInline svg={saveStartSVG} cleanup={['title']} />
+          </div>);
+        break;
+      case StepStateOption.ERROR:
+        middleBlock = (
+          <div className={classnames([styles.middleBlock, styles.middleSaveErrorBlock])}>
+            <SvgInline svg={saveErrorSVG} cleanup={['title']} />
+          </div>);
+        break;
+      default:
+        console.error('Error : something unexpected happened');
+        break;
+    }
+
+    const dailogActions = [{
+      className: this.props.isActionProcessing ? styles.processing : null,
+      label: intl.formatMessage(messages.saveButtonLabel),
+      primary: true,
+      disabled: this.props.isActionProcessing,
+      onClick: this.save
+    }];
+
+    return (
+      <Dialog
+        className={classnames([styles.component, 'SaveDialog'])}
+        title={intl.formatMessage(messages.title)}
+        actions={dailogActions}
+        closeOnOverlayClick={false}
+        onClose={this.props.cancel}
+        closeButton={<DialogCloseButton />}
+      >
+        <ProgressStepBlock progressInfo={this.props.progressInfo} />
+        {walletNameBlock}
+        {middleBlock}
+        <HelpLinkBlock progressInfo={this.props.progressInfo} />
+        <ErrorBlock progressInfo={this.props.progressInfo} error={this.props.error} />
+      </Dialog>);
+  }
+
+  save = async () => {
+    this.form.submit({
+      onSuccess: async (form) => {
+        const { walletName } = form.values();
+        this.props.submit(walletName);
+      },
+      onError: () => {},
```

Comment:
I would add a comment here why there *can’t be an error*. If there can be an error I would add it :D 🙏

Reply:
it will never be called because we are pre-validating the input,

![image](https://user-images.githubusercontent.com/19986226/49337250-d4572700-f653-11e8-9a3a-6b9e8883d3cb.png)


It's been used at 10 locations in the project and at 9 locations either it's `onError: () => {},`   or no `onError`